File: python_package/robot_simulator_2D/example_3DOF.py
import numpy as np
import matplotlib.pyplot as plt
import math
from robot_3DOF import robot_3DOF
##

length = [1, 1, 1]
bias = [0.2, 0.3]
robot = robot_3DOF('robot', length, bias, 2)

plt.ion()
plt.show()

N = 200
theta = np.linspace(0, 2*np.pi, N)
pitch = np.deg2rad(210)

traj = []
for i in range(N):
    traj.append([0.2 + 0.3 * math.cos(theta[i]), 0.3 + 0.3 * math.sin(theta[i])])
traj = np.array(traj)

for i in range(N):
    p3 = [traj[i, 0], traj[i, 1]]
    q = robot.ikine(p3, pitch + theta[i])
    robot.plot(q)
    plt.plot(traj[1:i, 0], traj[1:i, 1], linestyle='--', color='red', lw=2)
    plt.xlim((-1.5, 1.5))
    plt.ylim((-1.5, 1.5))
    plt.pause(0.01)
    ##
    rad = q.sum() - np.pi
    print(np.rad2deg(rad), '  ', robot.fkine(q))
    # print(robot.fkine(q))

File: matlab/package/robot_simulator_2D/programme_class/example_3DOF.m
clc;clear;close all
%%
robot = robot_3DOF(1,1,1,0.2,0.3,2);
%%
N = 200;
theta = linspace(0,2*pi,N);
pitch = deg2rad(210);
for i=1:N
    traj(i,1) = 0.2 + 0.3*cos(theta(i));
    traj(i,2) = 0.3 + 0.3*sin(theta(i));
    p3.x = traj(i,1);
    p3.y = traj(i,2);
    q = robot.ikine(p3,pitch + theta(i));
    the = sum(q) - pi;
    rad2deg(the)
    plot(traj(1:i,1),traj(1:i,2),'k--','linewidth',1.5);hold on
    robot.plot(q);hold off
    axis([-2 2 -2 2])
    grid on
    drawnow
end